src/main/java/org/apache/solr/benchmarks/BenchmarksMain.java

```diff
@@ -105,32 +108,124 @@ public static void runQueryBenchmarks(List<QueryBenchmark> queryBenchmarks, Stri
 		    for (int threads = benchmark.minThreads; threads <= benchmark.maxThreads; threads++) {
 		        QueryGenerator queryGenerator = new QueryGenerator(benchmark);
 		        HttpSolrClient client = new HttpSolrClient.Builder(baseUrl).build();
+		        String collection = collectionNameOverride==null? benchmark.collection: collectionNameOverride;
 		        ControlledExecutor controlledExecutor = new ControlledExecutor(threads,
 		                benchmark.duration,
 		                benchmark.rpm,
 		                benchmark.totalCount,
 		                benchmark.warmCount,
-		                getQuerySupplier(queryGenerator, client, collectionNameOverride==null? benchmark.collection: collectionNameOverride));
+		                getQuerySupplier(queryGenerator, client, collection));
 		        long start = System.currentTimeMillis();
+						List<SolrBenchQuery> generatedQueries;
 		        try {
-		            controlledExecutor.run();
+		            generatedQueries = controlledExecutor.run();
 		        } finally {
 		            client.close();
 		        }
 
+	        	int success = 0, failure = 0;
+
+		        if (StressMain.generateValidations) {
+		        	int totalDocsIndexed = getTotalDocsIndexed(baseUrl, collection);
```

Comment:
Since this is structured under `runQueryBenchmarks`, perhaps it shouldn't couple with docs indexed as well? This number could mean different things as well - The total docs in the collection or docs indexed during the benchmarking? As for "external" mode, it is valid to run ONLY query benchmarking (ie in the test plan, only test task is query) and start with a pre-populated collection. Therefore such number could change and probably shouldn't be captured here.

Perhaps we can safely remove this as the step of validation does not seem to validate on total # of docs either.

Reply:
My intention was to use that only to name the validation file, but not for any validation. I agree that validating with this wouldn't make sense in external mode.